.github/workflows/publish-c-apidocs.yml

```diff
@@ -0,0 +1,31 @@
+name: Publish C/C++ API Docs
+on:
+  workflow_dispatch
+
+jobs:
+  publish:
+    name: Publish C/C++ API docs
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install doxygen and dependencies
+        run: |
+          sudo apt update 
+          sudo apt-get install libclang-9-dev
+          sudo apt-get install libclang-cpp9
+          wget https://www.doxygen.nl/files/doxygen-1.9.2.linux.bin.tar.gz
+          tar xvzf doxygen-1.9.2.linux.bin.tar.gz
+      - name: Run doxygen
+        run: |
+          mkdir -p build/doxygen
+          cd docs/c_cxx
+          ../../doxygen-1.9.2/bin/doxygen
+      - name: Publish
+        uses: JamesIves/github-pages-deploy-action@releases/v3
+        with:
+          ACCESS_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
But where does the token come from?

Reply:
I don't believe this is actually required unless you are deploying to another repo. It defaults to the repository scoped token. I can try without.

> This option defaults to the repository scoped GitHub Token. However if you need more permissions for things such as deploying to another repository, you can add a Personal Access Token (PAT) here. 

From: https://github.com/JamesIves/github-pages-deploy-action#configuration-